src/Formats/NumpyDataTypes.h

```diff
@@ -0,0 +1,108 @@
+#include <cstddef>
+#include <Storages/NamedCollectionsHelpers.h>
+
+enum class NumpyDataTypeIndex
+{
+    Int8,
+    Int16,
+    Int32,
+    Int64,
+    UInt8,
+    UInt16,
+    UInt32,
+    UInt64,
+    Float32,
+    Float64,
+    String,
+    Unicode,
+};
+
+class NumpyDataType
+{
+public:
+    enum Endianness
+    {
+        LITTLE,
+        BIG,
+        NONE,
+    };
+
+    explicit NumpyDataType(Endianness endianess_) : endianess(endianess_) {}
+    virtual ~NumpyDataType() = default;
+
+    Endianness getEndianness() const { return endianess; }
+
+    virtual NumpyDataTypeIndex getTypeIndex() const = 0;
+    virtual size_t getSize() const = 0;
```

Comment:
Not sure if we need this method to be virtual. Now it's needed only for String/Unicode type, not all types (for ints and floats we don't need this method anymore).